oauth2/src/main/java/com/okta/spring/boot/oauth/env/OktaOAuth2PropertiesMappingEnvironmentPostProcessor.java

```diff
@@ -132,6 +144,9 @@ private PropertySource oktaStaticDiscoveryPropertySource(Environment environment
         Map<String, Object> properties = new HashMap<>();
         properties.put("spring.security.oauth2.resourceserver.jwt.issuer-uri", "${okta.oauth2.issuer}");
         properties.put("spring.security.oauth2.resourceserver.jwt.jwk-set-uri", "${okta.oauth2.issuer}/v1/keys");
+        properties.put("spring.security.oauth2.resourceserver.opaquetoken.client-id", "${okta.oauth2.clientId}");
+        properties.put("spring.security.oauth2.resourceserver.opaquetoken.client-secret", "${okta.oauth2.clientSecret}");
+        properties.put("spring.security.oauth2.resourceserver.opaquetoken.introspection-uri", "${okta.oauth2.issuer}/v1/introspect");
```

Comment:
What happens when the `client-id` is set but not `client-secret` ?

Reply:
resource server will start fine but the introspection wont work as it needs the secret to be sent.

Reply:
ok, so we need to make sure that runtime error does NOT occur.

We should probably skip the `isOpaque()` flag and just configure what is available, in cases where both are available the developer can use the `.jwt()` and `.opaqueToken()` methods, and we are not doing anything special or confusing.

Similaraly in the case of a "root/org" issuer, the JWT properties should NOT be configured.  This will alow the default Spring configuration to automatically use opaque tokens as that is the only one available.

In cases where a user tries to explicitly use `jwt()` in cases where it's not configured, it will fail at startup (which is ideal).

Reply:
makes sense.

Reply:
@bdemers I've pushed updates to code that will deal with the possible cases listed in the description section of this PR. Please let me know if I missed any use case.